### PR TITLE
Location-settings

### PR DESCRIPTION
**Pull request by Shilpi75**

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java [hunk 1]
```diff
@@ -79,6 +105,10 @@ private void startServiceIfAppEnabled() {
         appSwitch.setOnCheckedChangeListener((buttonView, isChecked) ->
                 onAppStatusChanged(isChecked));
         appSwitch.setChecked(isAppEnabled); // This will also trigger the onClickListener.
+        // If app is enabled, check for device's location settings.
+        if (appSwitch.isChecked()) {
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: use `isAppEnabled` here, conveys the meaning better and rename the method from checkSystemSettings to `checkLocationSettings`.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java [hunk 2]
```diff
@@ -257,4 +287,46 @@ public void onReceive(Context context, Intent intent) {
...
+                                try {
+                                    ResolvableApiException rae = (ResolvableApiException) e;
+                                    rae.startResolutionForResult(MainActivity.this,
+                                            REQUEST_CHECK_SETTINGS);
+                                } catch (IntentSender.SendIntentException sie) {
+                                    Log.i(TAG, "PendingIntent unable to execute request.");
+                                }
+                                break;
+                            case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+                                String errorMessage = "Location settings are inadequate, and " +
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Use string resource `error_location_switch_on_from_settings`

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java [hunk 3]
```diff
@@ -257,4 +287,46 @@ public void onReceive(Context context, Intent intent) {
...
+    /**
+     * Checks for required location settings according to setting's power saver preference.
+     */
+    public void checkSystemSettings() {
+        buildLocationsSettingsRequest();
+        mSettingsClient.checkLocationSettings(mLocationSettingsRequest)
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        int statusCode = ((ApiException) e).getStatusCode();
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Use lambda expressions.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java (hunk 2 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Not required.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java (hunk 3 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java (hunk 1 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Okay

**Pull request merged**